src/instructlab/data/generate.py

```diff
@@ -0,0 +1,218 @@
+    type=click.STRING,
+    default="",
+    help="TLS client certificate password.",
+)
+@click.option(
+    "--model-family",
+    help="Force model family to use when picking a generation template",
+)
+@click.pass_context
+def generate(
+    ctx,
+    model,
+    num_cpus,
+    num_instructions,
+    taxonomy_path,
+    taxonomy_base,
+    output_dir,
+    rouge_threshold,
+    quiet,
+    endpoint_url,
+    api_key,
+    yaml_rules,
+    chunk_word_count,
+    server_ctx_size,
+    tls_insecure,
+    tls_client_cert,
+    tls_client_key,
+    tls_client_passwd,
+    model_family,
+):
+    """Generates synthetic data to enhance your example data"""
+    # pylint: disable=C0415
+    # Local
+    from .generator.generate_data import generate_data
+    from .generator.utils import GenerateException
+    from instructlab.model.serve import ensure_server
+
+    server_process = None
+    server_queue = None
+    logger = logging.getLogger("TODO")
+    prompt_file_path = helpers.DEFAULT_PROMPT_FILE
+    if ctx.obj is not None:
+        logger = ctx.obj.logger
+        prompt_file_path = ctx.obj.config.generate.prompt_file
+
+    if endpoint_url:
+        api_base = endpoint_url
+    else:
+        # Third Party
+        import llama_cpp
+
+        if not llama_cpp.llama_supports_gpu_offload():
+            # TODO: check for working offloading. The function only checks
+            # for compile time defines like `GGML_USE_CUDA`.
+            click.secho(
+                "llama_cpp_python is built without hardware acceleration. "
+                "ilab generate will be very slow.",
+                fg="red",
+            )
+        try:
+            server_process, api_base, server_queue = ensure_server(
+                ctx.obj.logger,
+                ctx.obj.config.serve,
+                tls_insecure,
+                tls_client_cert,
+                tls_client_key,
+                tls_client_passwd,
+                model_family,
+            )
+        except Exception as exc:
+            click.secho(f"Failed to start server: {exc}", fg="red")
+            raise click.exceptions.Exit(1)
+        if not api_base:
+            api_base = ctx.obj.config.serve.api_base()
+    try:
+        click.echo(
+            f"Generating synthetic data using '{model}' model, taxonomy:'{taxonomy_path}' against {api_base} server"
+        )
+        generate_data(
+            logger=logger,
+            api_base=api_base,
+            api_key=api_key,
+            model_name=model,
+            num_cpus=num_cpus,
+            num_instructions_to_generate=num_instructions,
+            taxonomy=taxonomy_path,
+            taxonomy_base=taxonomy_base,
+            output_dir=output_dir,
+            prompt_file_path=prompt_file_path,
+            rouge_threshold=rouge_threshold,
+            console_output=not quiet,
+            yaml_rules=yaml_rules,
+            chunk_word_count=chunk_word_count,
+            server_ctx_size=server_ctx_size,
+        )
```

Comment:
This appears to have dropped TLS support from generate

Reply:
fixed